### ISSUE TITLE
Jellyfin source not respecting usersAllowed setting

### ISSUE THREAD

**Issue by TheFeelTrain** [FoxxMD/multi-scrobbler#196]
**Logs**

```
[2024-09-29 23:04:22.706 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Player State for  -> Astrophysics - Soft Goth <-- is being dropped because 'usersAllow does not include user Eriq
[2024-09-29 23:04:22.681 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Refreshing recently played
[2024-09-29 23:04:12.671 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Last activity was at 2024-09-29T23:03:42-07:00 | Next check interval: 10.00s
[2024-09-29 23:04:12.670 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] No new tracks discovered
[2024-09-29 23:04:12.670 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Player State for  -> Astrophysics - Soft Goth <-- is being dropped because 'usersAllow does not include user Eriq
[2024-09-29 23:04:12.643 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Refreshing recently played
[2024-09-29 23:04:02.630 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Last activity was at 2024-09-29T23:03:42-07:00 | Next check interval: 10.00s
[2024-09-29 23:04:02.629 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] No new tracks discovered
[2024-09-29 23:04:02.628 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Player State for  -> Astrophysics - Soft Goth <-- is being dropped because 'usersAllow does not include user Eriq
[2024-09-29 23:04:02.598 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Refreshing recently played
[2024-09-29 23:03:52.583 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Last activity was at 2024-09-29T23:03:42-07:00 | Next check interval: 10.00s
[2024-09-29 23:03:52.582 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] No new tracks discovered
[2024-09-29 23:03:52.582 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Player State for  -> Astrophysics - Soft Goth <-- is being dropped because 'usersAllow does not include user Eriq
[2024-09-29 23:03:52.546 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Refreshing recently played
[2024-09-29 23:03:42.534 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Last activity was at 2024-09-29T23:03:42-07:00 | Next check interval: 10.00s
[2024-09-29 23:03:42.525 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] No new tracks discovered
[2024-09-29 23:03:42.524 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Player State for  -> Astrophysics - Soft Goth <-- is being dropped because 'usersAllow does not include user Eriq
[2024-09-29 23:03:42.508 -0700] DEBUG  : [App] [Sources] [Jellyfin - TheJellyTrain] Refreshing recently played
```

**Versions:**

- multi-scrobbler: 0.8.4 docker
- Jellyfin 10.9.11

**Additional context**

```
[{
    "name": "TheJellyTrain",
    "enable": true,
    "clients": [],
    "data": {
        "url": "http://192.168.0.20:8096",
        "user": "Eriq",
        "apiKey": "redacted",
        "usersAllowed": ["Eriq"]
    },
    "options": {
        "logFilterFailure": "debug"
    }
}]
```
Same result with env variables and/or if the option is omitted even though it's supposed to be optional.

**Comment by TheFeelTrain** [comment]
Took me a minute but I figured out.

Not only does the code use `usersAllow` instead of `usersAllowed` like in the example it also has to be lowercase because of this line: https://github.com/FoxxMD/multi-scrobbler/blob/5adb83113667f9e1c1f9934aae89893530f00a5f/src/backend/sources/JellyfinApiSource.ts#L206

So this config works while the one in the original post doesn't.

```
[{
    "name": "TheJellyTrain",
    "enable": true,
    "clients": [],
    "data": {
        "url": "http://192.168.0.20:8096",
        "user": "Eriq",
        "apiKey": "redacted",
        "usersAllow": ["eriq"]
    },
    "options": {
        "logFilterFailure": "debug"
    }
}]
```

The `this.usersAllow.length > 0` also seems to not work as intended because it's checking even if the variable isn't set in the config which means it always be greater than 0.

**Comment by TheFeelTrain** [comment]
Took me a minute but I figured out.

The code uses `usersAllow` instead of `usersAllowed` like in the example and it also has to be lowercase because of this line: https://github.com/FoxxMD/multi-scrobbler/blob/5adb83113667f9e1c1f9934aae89893530f00a5f/src/backend/sources/JellyfinApiSource.ts#L206

So this config works while the one in the original post doesn't:

```
[{
    "name": "TheJellyTrain",
    "enable": true,
    "clients": [],
    "data": {
        "url": "http://192.168.0.20:8096",
        "user": "Eriq",
        "apiKey": "redacted",
        "usersAllow": ["eriq"]
    },
    "options": {
        "logFilterFailure": "debug"
    }
}]
```

The `this.usersAllow.length > 0` is also broken because it's returning true even when the variable isn't set in the config file. I'm not gonna dig for it but I assume it's getting set to something somewhere along the way where it shouldn't be.

**Comment by FoxxMD** [comment]
Thanks for catching this. It has been fixed in [0.8.5](https://github.com/FoxxMD/multi-scrobbler/releases/tag/0.8.5) so that the documentation matches the code by using `usersAllow` and other options everywhere.

> The this.usersAllow.length > 0 is also broken because it's returning true even when the variable isn't set in the config file. 

This is expected. If no config is set for `usersAllow` then it adds the authenticated user to the list as a convenience.

![image](https://github.com/user-attachments/assets/e944989d-e7a0-466a-acf5-32c659412f37)

**Comment by TheFeelTrain** [comment]
Thank you for the fast fix!

> This is expected. If no config is set for usersAllow then it adds the authenticated user to the list as a convenience.

Oh I see now. That's my bad, I only skimmed the code so I was missing the context for what that check was actually for 😅

I assumed it was checking if the config was set not a check to make sure the variable itself is set properly.

**Comment by FoxxMD** [comment]
These fixes are in [0.8.6](https://github.com/FoxxMD/multi-scrobbler/releases/tag/0.8.6) (`latest` docker image)
